source/as-promise.ts

```diff
@@ -44,21 +51,26 @@ const asPromise = options => {
 			response.body = data;
 
 			try {
-				for (const [index, hook] of Object.entries(options.hooks.afterResponse)) {
-					// eslint-disable-next-line no-await-in-loop
-					response = await hook(response, updatedOptions => {
-						updatedOptions = reNormalize(mergeOptions(options, {
-							...updatedOptions,
-							retry: 0,
-							throwHttpErrors: false
-						}));
-
-						// Remove any further hooks for that request, because we we'll call them anyway.
-						// The loop continues. We don't want duplicates (asPromise recursion).
-						updatedOptions.hooks.afterResponse = options.hooks.afterResponse.slice(0, index);
-
-						return asPromise(updatedOptions);
-					});
+				if (options.hooks && options.hooks.afterResponse) {
+					for (const [index, value] of Object.entries(options.hooks.afterResponse).entries()) {
```

Comment:
You're right that the statement was incorrect previously, but yours doesn't look correct either. I think it should be:

```suggestion
					for (const [index, hook] of options.hooks.afterResponse.entries()) {
```